src/circuit/gadget/ecc/chip/mul_fixed.rs

```diff
@@ -231,64 +187,41 @@ impl Config {
         &self,
         region: &mut Region<'_, pallas::Base>,
         offset: usize,
         scalar: &ScalarFixed,
-        base: OrchardFixedBases,
+        base: &F,
         coords_check_toggle: Selector,
     ) -> Result<(NonIdentityEccPoint, NonIdentityEccPoint), Error> {
         // Assign fixed columns for given fixed base
-        self.assign_fixed_constants::<NUM_WINDOWS>(region, offset, base, coords_check_toggle)?;
+        self.assign_fixed_constants::<F, NUM_WINDOWS>(region, offset, base, coords_check_toggle)?;
 
         // Initialize accumulator
-        let acc = self.initialize_accumulator(region, offset, base, scalar)?;
+        let acc = self.initialize_accumulator::<F, NUM_WINDOWS>(region, offset, base, scalar)?;
 
         // Process all windows excluding least and most significant windows
-        let acc = self.add_incomplete(region, offset, acc, base, scalar)?;
+        let acc = self.add_incomplete::<F, NUM_WINDOWS>(region, offset, acc, base, scalar)?;
 
         // Process most significant window using complete addition
-        let mul_b = self.process_msb::<NUM_WINDOWS>(region, offset, base, scalar)?;
+        let mul_b = self.process_msb::<F, NUM_WINDOWS>(region, offset, base, scalar)?;
 
         Ok((acc, mul_b))
     }
 
-    fn assign_fixed_constants<const NUM_WINDOWS: usize>(
+    fn assign_fixed_constants<F: FixedPoint<pallas::Affine>, const NUM_WINDOWS: usize>(
         &self,
         region: &mut Region<'_, pallas::Base>,
         offset: usize,
-        base: OrchardFixedBases,
+        base: &F,
         coords_check_toggle: Selector,
     ) -> Result<(), Error> {
-        let mut constants = None;
-
-        let build_constants = || match base {
-            OrchardFixedBases::ValueCommitV => {
-                assert_eq!(NUM_WINDOWS, constants::NUM_WINDOWS_SHORT);
-                let base = ValueCommitV::get();
-                (
-                    base.lagrange_coeffs_short.0.as_ref().to_vec(),
-                    base.z_short.0.as_ref().to_vec(),
-                )
-            }
-            OrchardFixedBases::Full(base) => {
-                assert_eq!(NUM_WINDOWS, constants::NUM_WINDOWS);
-                let base: OrchardFixedBase = base.into();
-                (
-                    base.lagrange_coeffs.0.as_ref().to_vec(),
-                    base.z.0.as_ref().to_vec(),
-                )
-            }
-            OrchardFixedBases::NullifierK => {
-                assert_eq!(NUM_WINDOWS, constants::NUM_WINDOWS);
-                let base: OrchardFixedBase = NullifierK.into();
-                (
-                    base.lagrange_coeffs.0.as_ref().to_vec(),
-                    base.z.0.as_ref().to_vec(),
-                )
-            }
-        };
+        let lagrange_coeffs = base.lagrange_coeffs();
+        assert_eq!(lagrange_coeffs.len(), NUM_WINDOWS);
+
+        let z = base.z();
+        assert_eq!(z.len(), NUM_WINDOWS);
```

Comment:
I checked locally, and indeed this would have caused a regression:
```
proving/bundle/1        time:   [3.8017 s 3.8160 s 3.8325 s]                              
                        change: [+12.469% +13.295% +14.165%] (p = 0.00 < 0.05)
                        Performance has regressed.
```